### ISSUE TITLE
Builtin mapping of String <-> Duration/Period/Instant

### ISSUE THREAD

**Issue by twentylemon** [mapstruct/mapstruct#1656]
A fair chunk of the java.time objects have builtin mappings to and from String. Duration and Period also has a standard string format, the docs describe it a bit.

* [Duration#parse](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#parse-java.lang.CharSequence-)
* [Duration#toString](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#toString--)
* [Period#parse](https://docs.oracle.com/javase/8/docs/api/java/time/Period.html#parse-java.lang.CharSequence-)
* [Period#toString](https://docs.oracle.com/javase/8/docs/api/java/time/Period.html#toString--)
* [Instant](https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html#toString--) is similar, using `DateTimeFormatter.ISO_INSTANT`

However, MapStruct doesn't support these conversions [implicitly](http://mapstruct.org/documentation/dev/reference/html/#implicit-type-conversions). Would be a nice to have, out of the box, to fit with the rest of java.time mappings.

**Comment by filiphr** [comment]
Agreed it would be good to have support for the missing `java.time` objects. Support for `Instant` has been added in https://github.com/mapstruct/mapstruct/issues/1595 which is within the 1.3.0.Beta2 release.

I'll rename the issue to remove the `Instant` from it.

Is this something that you would like to look into?

**Comment by filiphr** [comment]
I was wrong about the `Instant`. With #1595 we added support for `java.util.Date` to / from `java.time.Instant`

**Comment by twentylemon** [comment]
I could try taking a look -- though it's a very busy time of year. Judging by #1622 it's a fairly easy change.
